velox/type/TimestampConversion.h

```diff
@@ -107,6 +107,38 @@ Status daysSinceEpochFromWeekDate(
     int32_t dayOfWeek,
     int64_t& out);
 
+/// Computes the signed number of days since the Unix epoch (1970-01-01). To
+/// align with Spark's SimpleDateFormat behavior, this function offers two
+/// modes: lenient and non-lenient. For non-lenient mode, it returns an error
+/// status if the date is invalid. For lenient mode, it accepts a wider range of
+/// arguments.
+/// @param year Year. For non-lenient mode, it should be in the range [1,
+/// 292278994]. e.g: 1996, -2000. For lenient mode, values outside this range
+/// could result in overflow.
+/// @param month Month of year. For non-lenient mode, it should be in the range
+/// [1, 12]. For example, 1 is January, 7 is July. For lenient mode, values
+/// greater than 12 wrap around to the start of the year, and values less than 1
+/// count backward from December. For example, 13 corresponds to January of the
+/// following year and -1 corresponds to November of the previous year.
+/// @param weekOfMonth Week of the month. For non-lenient mode, it should be in
+/// the range [1, 6]. For example, 1 is 1st week, 3 is 3rd week. For lenient
```

Comment:
> the range [1, 6]

Is this the right range. Is there a month with 6 weeks? For value '5', some months doesn't have week 5. Is the behavior in that case same as in 'lenient' mode. Would be nice to clarify.